src/pytesmo/validation_framework/data_averaging.py

```diff
@@ -0,0 +1,423 @@
+        hours: int
+            window to perform the temporal matching
+
+        Returns
+        -------
+        matched: pd.DataFrame
+            dataframe with temporally matched timeseries
+        """
+        if method == 'common':
+            matched = pd.concat(to_match, axis=1, join="inner").dropna()
+
+        elif method == 'rescale':
+            # get time series with most points
+            for n, df in enumerate(to_match):
+                if df is None:
+                    continue
+                points = len(df.dropna())
+                if n == 0:
+                    ref = df
+                if len(ref.dropna()) >= points:
+                    continue
+                else:
+                    ref = df
+
+            to_match = pd.concat(to_match, axis=1)
+            matched = tcoll(
+                ref,
+                to_match,
+                pd.Timedelta(hours, "H"),
+                dropna=True
+            )
+
+        return matched
+
+    @staticmethod
+    def tstability_filter(
+            df,
+            r_min=0.6,
+            see_max=0.05,
+            min_n=4,
+            **kwargs,
+    ) -> pd.DataFrame:
+        """
+        Uses time stability concepts to filter point-measurements (pms). Determines wether the upscaled measurement
+        based on a simple average of all the pms is in sufficient agreement with each pm, and if not eliminates pm
+        from the pool.
+
+        Parameters
+        ----------
+        df: pd.DataFrame
+            temporally matched DataFrame
+        r_min: float
+            lower threshold for correlation (Pearson) between upscaled and pm
+        see_max: float
+            upper threshold for standard error of estimate
+        min_n: int
+            minimum number of pms to perform the filtering
+
+        Returns
+        -------
+        filtered: pd.DataFrame
+            filtered input
+        """
+        if len(df.columns) < min_n:
+            return df
+
+        # get a trivial (average) upscale estimate
+        estimate = df.mean(axis=1)
+
+        filter_out = []
+        for n, pm in enumerate(df):
+            pm_values = df[pm]
+            regr = linregress(
+                estimate,
+                pm_values
+            )
+            if regr.rvalue < r_min or regr.intercept_stderr > see_max:
+                filter_out.append(pm)
+
+        filtered = df.drop(filter_out)
+        # todo: This could be done iteratively to calibrate the thresholds
+        if len(filtered.columns) < 2:
+            warnings.warn(
+                "The filtering options are too strict. Returning entire dataframe ..."
+            )
+            return df
+
+        return filtered
+
+    def upscale(self, df, method='average', **kwargs) -> pd.DataFrame:
+        """
+        Return the upscaled Dataframe with the specified method
+
+        **New upscaling methods can be specified here in the lut**
```

Comment:
Move this comment outside of the docstring, it's only relevant for developers. I would just put it in the line above `up_function`. Also, I'm not sure how universally "lut" is understood as look-up table, took me some time to figure it out, so maybe just write it out in the comment.